src/main/host/process.rs

```diff
@@ -510,6 +528,71 @@ impl RunnableProcess {
         host.schedule_task_with_delay(task, SimulationTime::ZERO);
     }
 
+    pub fn new_forked_process(
+        &self,
+        host: &Host,
+        new_tgl: RootedRc<RootedRefCell<Thread>>,
+    ) -> RootedRc<RootedRefCell<Process>> {
+        let new_tgl_tid;
+        let native_pid;
+        {
+            let new_tgl = new_tgl.borrow(host.root());
+            new_tgl_tid = new_tgl.id();
+            native_pid = new_tgl.native_pid();
+        }
+        let pid = ProcessId::from_thread_group_leader_tid(new_tgl_tid);
```

Comment:
Does the thread itself keep track of if it's a thread group leader or not? Could we assert here that the tid does refer to a thread group leader so that we know `ProcessId::from_thread_group_leader_tid` is a valid operation?